federationtypes.go

```diff
@@ -370,8 +370,14 @@ func (r RespState) Events() ([]Event, error) {
 	return result, nil
 }
 
+// MissingAuthEventHandler is a function signature which can be provided to
+// RespState.Check(), RespSendJoin.Check() and checkAllowedByAuthEvents. If
+// an auth event comes up missing during the checks, the MissingAuthEventHandler
+// will be called (if provided) to try and retrieve the missing event.
+type MissingAuthEventHandler func(eventID string) (*Event, error)
```

Comment:
You'll probably need the room version as per https://github.com/matrix-org/gomatrixserverlib/blob/master/backfill.go#L26

Reply:
Have added it for good measure, although the caller probably already knows it.